src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -2436,6 +2436,240 @@ void PerimeterGenerator::process_no_bridge(Surfaces& all_surfaces, coord_t perim
     }
 }
 
+// ORCA:
+// Inner Outer Inner wall ordering mode perimeter order optimisation functions
+
+/**
+ * @brief Converts the junctions of an ExtrusionLine to a vector of Points.
+ *
+ * @param line The input ExtrusionLine containing junctions to be converted.
+ * @return std::vector<Point> A vector containing points extracted from the junctions of the input ExtrusionLine.
+ */
+std::vector<Point> extrusionLineToPointsIOI(const Arachne::ExtrusionLine& line) {
+    std::vector<Point> points;
+    for (const auto& junction : line.junctions) {
+        points.push_back(junction.p);
+    }
+    return points;
+}
+
+/**
+ * @brief Calculates the squared distance between two points.
+ *
+ * @param a The first point.
+ * @param b The second point.
+ * @return double The squared distance between the two points.
+ */
+double squaredDistance(const Point& a, const Point& b) {
+    return (a.x() - b.x()) * (a.x() - b.x()) + (a.y() - b.y()) * (a.y() - b.y());
+}
+
+/**
+ * @brief Calculates the squared distance between a point and a line segment.
+ *
+ * @param p The point.
+ * @param v The starting point of the line segment.
+ * @param w The ending point of the line segment.
+ * @return double The squared distance between the point and the line segment.
+ */
+double squaredDistanceToSegment(const Point& p, const Point& v, const Point& w) {
+    double l2 = squaredDistance(v, w);
+    if (l2 == 0.0) return squaredDistance(p, v);
+    double t = std::max(0.0, std::min(1.0, ((p.x() - v.x()) * (w.x() - v.x()) + (p.y() - v.y()) * (w.y() - v.y())) / l2));
+    Point projection{v.x() + t * (w.x() - v.x()), v.y() + t * (w.y() - v.y())};
+    return squaredDistance(p, projection);
+}
+
+/**
+ * @brief Calculates the minimum distance between any two polylines defined by sets of points.
+ *
+ * @param A The first set of points defining a polyline.
+ * @param B The second set of points defining a polyline.
+ * @return double The minimum distance between the two polylines.
+ */
+double minimumDistanceIOI(const std::vector<Point>& A, const std::vector<Point>& B) {
```

Comment:
Can we make it a member function of MultiPoint class?

Reply:
Working on it :)

Reply:
This function has now moved to the MultiPoint class. Uploading new commits shortly.